tests/integration/targets/aws_sqs_queue/tasks/main.yml

```diff
@@ -0,0 +1,156 @@
+        #  assert:
+        #    that:
+        #      - role_info.iam_roles[0].tags.Name == "{{ resource_id }}"
+        #  when: number_of_roles == 1
+      vars:
+        number_of_roles: 0
+
+    - name: Terraform in present check mode
+      cloud.terraform.terraform:
+        project_path: "{{ test_basedir }}"
+        state: present
+        force_init: true
+        variables:
+          cloud_terraform_integration_id: "{{ resource_id }}"
+      register: terraform_result
+      check_mode: true
+
+    - assert:
+        that:
+          - terraform_result is not failed
+          - terraform_result is changed
+
+    - <<: *verification
+      vars:
+        number_of_roles: 0
+
+    - name: Terraform in present non-check mode
+      cloud.terraform.terraform:
+        project_path: "{{ test_basedir }}"
+        state: present
+        force_init: true
+        variables:
+          cloud_terraform_integration_id: "{{ resource_id }}"
+      register: terraform_result
+      check_mode: false
+
+    - assert:
+        that:
+          - terraform_result is not failed
+          - terraform_result is changed
+
+    - <<: *verification
+      vars:
+        number_of_roles: 1
+
+    - name: Terraform in present non-check mode (idempotency)
+      cloud.terraform.terraform:
+        project_path: "{{ test_basedir }}"
+        state: present
+        force_init: true
+        variables:
+          cloud_terraform_integration_id: "{{ resource_id }}"
+      register: terraform_result
+      check_mode: false
+
+    - assert:
+        that:
+          - terraform_result is not failed
+          - terraform_result is not changed
+
+    - <<: *verification
+      vars:
+        number_of_roles: 1
+
+    - name: Terraform in absent check mode
+      cloud.terraform.terraform:
+        project_path: "{{ test_basedir }}"
+        state: absent
+        force_init: true
+        variables:
+          cloud_terraform_integration_id: "{{ resource_id }}"
+      register: terraform_result
+      check_mode: true
+
+    - assert:
+        that:
+          - terraform_result is not failed
+          - terraform_result is changed
+
+    - <<: *verification
+      vars:
+        number_of_roles: 1
+
+    - name: Terraform in absent non-check mode
+      cloud.terraform.terraform:
+        project_path: "{{ test_basedir }}"
+        state: absent
+        force_init: true
+        variables:
+          cloud_terraform_integration_id: "{{ resource_id }}"
+      register: terraform_result
+      check_mode: false
+
+    - assert:
+        that:
+          - terraform_result is not failed
+          - terraform_result is changed
+
+    - <<: *verification
+      vars:
+        number_of_roles: 0
+
+  # Clean up all integration test resources
+  always:
+    - name: Fetch IAM role info
+      amazon.aws.iam_role_info:
+        name: "{{ test_role_name }}"
+      register: role_info
+      ignore_errors: true
```

Comment:
This shouldn't be necessary. There would be only one role with this name so you can just delete the role by name without the info task.